torch/distributed/elastic/rendezvous/api.py

```diff
@@ -35,6 +38,61 @@ class RendezvousGracefulExitError(RendezvousError):
     Exception is a mechanism to exit the stack, however does not mean a failure.
     """
 
+@dataclass
+class RendezvousStoreInfo:
+    """Store address and port that can be used to bootstrap trainer distributed comms"""
+    MASTER_ADDR_KEY: ClassVar[str] = "MASTER_ADDR"
+    MASTER_PORT_KEY: ClassVar[str] = "MASTER_PORT"
+    master_addr: str
+    master_port: int
+
+    @staticmethod
+    def build(rank: int, store: Store) -> "RendezvousStoreInfo":
```

Comment:
I'm confused about the order of operations.

I thought the build() path was used when we did not have a store to share from the elastic later.

Actually I guess it must be the following- we always have a tcpstore at the elastic layer, so we can use it to share information. But we call build() when we don't want to share that store with the trainer, and instead we just use the elastic's store for exchanging port info so the trainer can create its own store?

Reply:
Yes. Build here is a factory method of the `RendezvousStoreInfo` that will define `master_addr/master_port` env variables that trainer code can use to start a new store server.